src/Wpdb/AbstractBaseWpdbSelectResourceModel.php

```diff
@@ -399,6 +393,42 @@ protected function _setMapFactory(MapFactoryInterface $mapFactory)
         $this->mapFactory = $mapFactory;
     }
 
+    /**
+     * Creates the result set from raw record data sets.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|stdClass|Traversable $rawResults The list of raw record data sets.
+     *
+     * @return MapInterface[]|stdClass|Traversable A list of maps, each containing data for a record.
```

Comment:
Array of arrays is allowed.

Reply:
[See here](https://github.com/Dhii/cqrs-resource-model-interface/blob/develop/src/SelectCapableInterface.php#L33).

Reply:
Confused return type with param type.